IndexedDB/idbcursor-request.any.js

```diff
@@ -0,0 +1,42 @@
+// META: script=support.js
+
+function cursorRequestTest({ useIndex, useKeyCursor }) {
+  indexeddb_test(
+    (t, db) => {
+      const objStore = db.createObjectStore("my_objectstore");
+      objStore.add("data",  1);
+      objStore.createIndex("my_index", "");
+    },
+    (t, db) => {
+      const tx = db.transaction("my_objectstore");
+      let source = tx.objectStore("my_objectstore");
+      if (useIndex) source = source.index('my_index');
+      const req = useKeyCursor ? source.openKeyCursor() : source.openCursor();
+      let cursor;
+
+      req.onsuccess = t.step_func(() => {
+        cursor = req.result;
+        assert_equals(cursor.request, req, 'cursor.request');
+        assert_readonly(cursor, 'request');
+      });
+
+      req.transaction.oncomplete = t.step_func(() => {
+        setTimeout(t.step_func(() => {
```

Comment:
Use `t.step_timeout` per https://web-platform-tests.org/writing-tests/testharness-api.html

Reply:
`step_timeout` is used for "wait a bit" cases, and to allow multipliers to be applied to the duration, whereas I'm using `setTimeout` it to wait until the next task. The timeout is 0 so multipliers don't do anything.

Reply:
ack. (maybe we should codify that as `t.step_task` .... )